File: app/src/main/java/com/kunzisoft/keepass/magikeyboard/receiver/LockBroadcastReceiver.java
package com.kunzisoft.keepass.magikeyboard.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public abstract class LockBroadcastReceiver extends BroadcastReceiver {

    public static final String LOCK_ACTION = "com.kunzisoft.keepass.LOCK";

    @Override
    public void onReceive(Context context, Intent intent) {
        String action = intent.getAction();
        if (action != null
                && action.equals(LOCK_ACTION)) {
            onReceiveLock(context, intent);
        }
    }

    public abstract void onReceiveLock(Context context, Intent intent);

}

File: app/src/main/java/com/kunzisoft/keepass/magikeyboard/receiver/LockBroadcastReceiver.kt
package com.kunzisoft.keepass.magikeyboard.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

abstract class LockBroadcastReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        val action = intent.action
        if (action != null && action == LOCK_ACTION) {
            onReceiveLock(context, intent)
        }
    }

    abstract fun onReceiveLock(context: Context, intent: Intent)

    companion object {

        const val LOCK_ACTION = "com.kunzisoft.keepass.LOCK"
    }

}
